tensornetwork/network_components_test.py

```diff
@@ -433,21 +469,29 @@ def test_copy_node_load(tmp_path, backend):
     node_group = node_file.create_group('node_data')
     node_group.create_dataset('signature', data=node.signature)
     node_group.create_dataset('name', data=node.name)
+    node_group.create_dataset(
+        'copy_node_dtype', data=np.dtype(node.copy_node_dtype).name)
```

Comment:
Why should the `copy_node_dtype` be different from the dtype of the node? Seems redundant.

Reply:
that's a technical detail: the current implementation of `CopyNode` instantiates
`CopyNode._tensor` when calling `CopyNode.tensor` property. This instantiation
is always done in `numpy`, irrespective of the actual backend, and the `copy_node_dtype` is used as dtype if this instantiation. `CopyNode.tensor` then calls `backend.convert_to_tensor(numpy_tensor)` on the numpy tensor to create the actual tensor object returned by `CopyNode.tensor`. The dtype of the `numpy_tensor` has to match 
the dtype of `backend`, and the lazy solution to this problem is to let the use pass this numpy dtype. The better thing would be to determine the numpy-dtype from `backend.dtype`. That's a bit ugly to code but not a real problem. I'll add it